### PR TITLE
Rework introduction

### PR DESCRIPTION
**Pull request by Nicoretti**
* Add "Who Embedded Rust is For" section
* Rework "Who This Book is for section" to be more inclusive
  but still state what knowlege is important to get more out of the book

References:
* https://github.com/rust-embedded/book/issues/3

### CODE REVIEW

File: src/intro/introduction.md [hunk 1]
```diff
@@ -41,15 +50,19 @@ This book assumes the following:
...
-If you are not yet comfortable with Rust, we highly suggest completing the [Rust
-Book] before reading this book.
-
-[Rust Book]: https://doc.rust-lang.org/book/second-edition
+### Other Resources
+If you are unfamiliar with anything mentioned above or if you want more information about a specific topic mentioned in this book you might find some of these resources helpful.
 
-If you are not yet comfortable with embedded systems, we highly suggest checking
-out our [other resources] before attempting to read this book.
+| Topic        | Resource | Description |
```
Comment (korken89) [inline src/intro/introduction.md]: This table does not render properly

File: src/intro/introduction.md [hunk 2]
```diff
@@ -3,6 +3,10 @@
 Welcome to The Embedded Rust Book: An introductory book about using the Rust
 Programming Language on "Bare Metal" embedded systems, such as Microcontrollers.
 
+## Who Embedded Rust is For
+Embedded rust is for everyone who wants to do embedded programming backed by the higher-level concepts and safety guarantees the rust language provides.
+(See also [Who Rust Is For](https://doc.rust-lang.org/book/2018-edition/ch00-00-introduction.html)
```
Comment (korken89) [inline src/intro/introduction.md]: Missing ending )

File: src/intro/introduction.md [hunk 3]
```diff
@@ -24,8 +28,13 @@ particular architecture and explains details particular to this architecture
 where required.
 
 ## Who This Book is For
+This book caters towards people with either a solid embedded background or a solid rust background, however we assume
```
Comment (therealprof) [inline src/intro/introduction.md]: "solid" is quite a strong word IMHO, I'd tone this down a little bit so people don't give up at this point already. How about "some"? Also the repetition of "solid" and "background" could be avoided.

File: src/intro/introduction.md (hunk 1 above)
Comment (Nicoretti) [reply src/intro/introduction.md]: Thx, did not see that one.

File: src/intro/introduction.md (hunk 2 above)
Comment (Nicoretti) [reply src/intro/introduction.md]: *thumbs up* thx again

File: src/intro/introduction.md (hunk 3 above)
Comment (Nicoretti) [reply src/intro/introduction.md]: I'll change that, still leaving out one some/solid sounds kinda odd to me (without restructuring the sentence).

**Comment by thejpster** [comment]
I didn't want to block this with a review, but Rust has a capital R if you get the chance to make a quick update :)

File: src/intro/introduction.md [hunk 4]
```diff
@@ -41,15 +50,18 @@ This book assumes the following:
...
+
+| Topic        | Resource | Description |
+|--------------|----------|-------------|
+| Rust         | [Rust Book 2018 Edition](https://doc.rust-lang.org/book/2018-edition/index.html) | If you are not yet comfortable with Rust, we highly suggest reading the this book. |
+| Rust         | [Rust Book Second Edition](https://doc.rust-lang.org/book/second-edition) | - |
+| Rust, Embedded | [Embedded Rust Bookshelf](https://rust-embedded.github.io/bookshelf/) | - |
+| Rust, Embedded | [Embedonomicon](https://rust-embedded.github.io/embedonomicon/) | The nitty gritty details when doing embedded programming in rust. |
+| Interrupts | [Interrupt](https://en.wikipedia.org/wiki/Interrupt) | - |
+| Memory Mapped IO/Peripherals | [Memor-mapped I/O](https://en.wikipedia.org/wiki/Memory-mapped_I/O) | - |
+| SPI, UART, RS232, USB, I2C, TTL | [Stack Excahnge about SPI, UART, ...](https://electronics.stackexchange.com/questions/37814/usart-uart-rs232-usb-spi-i2c-ttl-etc-what-are-all-of-these-and-how-do-th) | - |
```
Comment (therealprof) [inline src/intro/introduction.md]: "Stack Excahnge" -> "Stack Exchange"

File: src/intro/introduction.md [hunk 5]
```diff
@@ -3,6 +3,10 @@
 Welcome to The Embedded Rust Book: An introductory book about using the Rust
 Programming Language on "Bare Metal" embedded systems, such as Microcontrollers.
 
+## Who Embedded Rust is For
+Embedded rust is for everyone who wants to do embedded programming backed by the higher-level concepts and safety guarantees the rust language provides.
```
Comment (therealprof) [inline src/intro/introduction.md]: I agree with @thejpster: Please go through the instances where you used "rust" and replace them with "Rust".

**Comment by Nicoretti** [comment]
Hi Guys,

Thx for the reviews,
I’m currently on a trip and I do not have any device except my Phone with me. I’ll fix it as soon as I am Back  ~17/18 Oct.

greez
Nico

File: src/intro/introduction.md [hunk 6]
```diff
@@ -41,15 +50,18 @@ This book assumes the following:
...
-
-[other resources]: https://rust-embedded.github.io/bookshelf/
+### Other Resources
+If you are unfamiliar with anything mentioned above or if you want more information about a specific topic mentioned in this book you might find some of these resources helpful.
+
+| Topic        | Resource | Description |
+|--------------|----------|-------------|
+| Rust         | [Rust Book 2018 Edition](https://doc.rust-lang.org/book/2018-edition/index.html) | If you are not yet comfortable with Rust, we highly suggest reading the this book. |
+| Rust         | [Rust Book Second Edition](https://doc.rust-lang.org/book/second-edition) | - |
+| Rust, Embedded | [Embedded Rust Bookshelf](https://rust-embedded.github.io/bookshelf/) | - |
```
Comment (adamgreig) [inline src/intro/introduction.md]: Please could you also add a link to the [embedded FAQ](https://rust-embedded.github.io/bookshelf/faq.html)?

File: src/intro/introduction.md [hunk 7]
```diff
@@ -41,15 +50,19 @@ This book assumes the following:
...
+| Rust, Embedded | [Embedonomicon](https://rust-embedded.github.io/embedonomicon/) | The nitty gritty details when doing embedded programming in Rust. |
+| Rust, Embedded | [embedded FAQ](https://rust-embedded.github.io/bookshelf/faq.html) | Frequently asked questions about rust in an embedded context |
+| Interrupts | [Interrupt](https://en.wikipedia.org/wiki/Interrupt) | - |
+| Memory Mapped IO/Peripherals | [Memor-mapped I/O](https://en.wikipedia.org/wiki/Memory-mapped_I/O) | - |
```
Comment (therealprof) [inline src/intro/introduction.md]: Memor-mapped I/O should be **Memory-mapped I/O**

File: src/intro/introduction.md [hunk 8]
```diff
@@ -41,15 +50,19 @@ This book assumes the following:
...
+| SPI, UART, RS232, USB, I2C, TTL | [Stack Excahnge about SPI, UART, ...](https://electronics.stackexchange.com/questions/37814/usart-uart-rs232-usb-spi-i2c-ttl-etc-what-are-all-of-these-and-how-do-th) | - |
```
Comment (therealprof) [inline src/intro/introduction.md]: Stack Excahnge should be **Stack Exchange**

**Pull request merged**